### ISSUE TITLE
Should we resurrect activation logs changes which are now in a separate repo?

### ISSUE THREAD

**Issue by rabbah** [nimbella/aio-cli-plugin-runtime#8]
The upstream repo is undergoing some refactoring. We haven't picked up changes since June and there are now some conflicts for us to work through. 

Will keep an eye on https://github.com/adobe/aio-cli-plugin-runtime/issues/191.

**Comment by rabbah** [comment]
This is a list of commits in our fork that are upstreamable but need a nod from Adobe. A check mark indicates a PR is active upstream.

- [x] [parameterize the "list" limits](https://github.com/nimbella/aio-cli-plugin-runtime/commit/9de99c1e8e94389befb990d0142f1c387a2acac8)
- [x] [fetch multiple activation logs at once](https://github.com/nimbella/aio-cli-plugin-runtime/commit/2e45c318bea717f3c06c19ee3c7d19f1bce8f577)
- [ ] [fetch multiple results at once](https://github.com/nimbella/aio-cli-plugin-runtime/commit/12130ea37da303bf2d064c2f1dc521f060ed0114)
- [ ] [make invoke synchronous by default](https://github.com/nimbella/aio-cli-plugin-runtime/commit/72ffa7507d89842eef5b4fb4f233d0f058daae67)
- [ ] [filter get last activation by name](https://github.com/nimbella/aio-cli-plugin-runtime/commit/582ecb2ec8170f0daf48fe2f95e4e123eef2f7bb)
- [ ] [add log banner](https://github.com/nimbella/aio-cli-plugin-runtime/commit/7b355f2054714a6c458103bdf75f24c5f97780b4)
- [x] [fix triggers in activation list](https://github.com/nimbella/aio-cli-plugin-runtime/commit/39be01b81d3315aa5c6b9c7706f3278aed3034c3)
- [ ] [add typescript kind](https://github.com/nimbella/aio-cli-plugin-runtime/commit/887912c9213f9bd0557e8632e713a839ce073b72)
- [ ] [add --result to activation get](https://github.com/nimbella/aio-cli-plugin-runtime/commit/cd6b702572c3984a790ba311d2464edd81f67f24)
- [ ] [allow web actions to be created with a secret](https://github.com/nimbella/aio-cli-plugin-runtime/commit/867f5895ef98dc0241c486a05b3f9350a8b0122e)
- [ ] [reformat entity list commands](https://github.com/nimbella/aio-cli-plugin-runtime/commit/cd694389a4c640bd581a92727b6490f8921a1b58)
- [ ] [add human readable date to assets](https://github.com/nimbella/aio-cli-plugin-runtime/commit/d82fc45229afcb8d10e05f5dc5e351f6cf2ddd3a)
- [ ] [omit table headers when lists are empty](https://github.com/nimbella/aio-cli-plugin-runtime/commit/e31a72cbe065f5146aa8410a7687bd281583bddc)
- [ ] [show request id in error when available](https://github.com/nimbella/aio-cli-plugin-runtime/commit/a6c3d8a2013cae5f7bc709566228a67c4a1d5a2b)
- [ ] [if no activation id is given assume --last](https://github.com/nimbella/aio-cli-plugin-runtime/commit/f1be9edcdefb4d827e9a50af2140a8bf5bf73b59)
- [ ] [show code when asked](https://github.com/nimbella/aio-cli-plugin-runtime/commit/d325cc1b427baaf4e5c110cea3118e58a699d45e)
- [ ] [remove namespace from lists](https://github.com/nimbella/aio-cli-plugin-runtime/commit/1e4d39b7240dd5eac3e52f7b4c708c0d3e1141d6)
- [ ] [add activation column for sequences and workflows](https://github.com/nimbella/aio-cli-plugin-runtime/commit/25f0dd29382ab172ad1d780028b350b22116064d)
- [ ] [allow action create to omit name, use file name instead](https://github.com/nimbella/aio-cli-plugin-runtime/commit/f99a5c8322dafdb3a76a48ee06e168489f2b15e0)
- [ ] [show timeouts in activation list](https://github.com/nimbella/aio-cli-plugin-runtime/commit/62b999388e9eb16bb6c98eec106b1f4757b15af2)
- [x] [add --count for lists](https://github.com/nimbella/aio-cli-plugin-runtime/commit/b786569ac590ec67264f3dc4844d2eadf79e28d4)
- [ ] [add --native action support for binaries](https://github.com/nimbella/aio-cli-plugin-runtime/commit/45c28f2f7316e5751d0508976e8cd75ec5a3b455)
- [ ] [add final annotations when making web actions](https://github.com/nimbella/aio-cli-plugin-runtime/commit/9028cb124a16ad3e91ae9b1c720c02207c062e34)
- [ ] [add version to action list](https://github.com/nimbella/aio-cli-plugin-runtime/commit/cecb30a5c1d30a24e6c91331bc69b63d677bbe39)
- [ ] [allow some slicing of log lines and silence header](https://github.com/nimbella/aio-cli-plugin-runtime/commit/bd667ff2d8b4415dfd5ce2eec3d167fa3ba59d33)
- [ ] [format log lines as JSON if possible](https://github.com/nimbella/aio-cli-plugin-runtime/commit/a6b3cf31ef9bba68967665460ff8468e51d34465)
- [ ] [allow get logs --last to skip](https://github.com/nimbella/aio-cli-plugin-runtime/commit/e73188cb61a6bf29df883e515a27c64318b06d7c)

**Comment by rabbah** [comment]
Examples of new list format:
```
> nim action list
Datetime        Access   Kind       Version  Actions                                           
10/21 11:10:54  web      python:3ai 0.0.3    school/import                                     
10/18 14:40:59  web      java:8     0.0.1    qr-java  
```

```
> nim activation list --limit 10
Datetime        Status   Kind       Version    Activation ID                    Start Wait Init Duration Entity        
10/21 11:16:55  error💥  python:3ai 0.0.3      0e73665b18a045fbb3665b18a075fb73 cold  918  3728 3728ms   school/import 
10/14 10:35:35  success  python:3   0.0.1      34153295f3544eeb953295f3549eeb27 warm  144  0    2ms      t             
10/14 10:35:27  error    python:3   0.0.1      7b6dba1228604249adba122860c24995 cold  549  114  116ms    t             
10/11 20:17:31  success  java:8     0.0.9      c100c7821bd84d1080c7821bd83d1026 cold  510  410  980ms    qr-java       
10/11 20:15:28  success  java:8     0.0.8      6c3277e6cfa94aeab277e6cfa9daea34 cold  524  461  1071ms   qr-java       
10/11 20:14:46  error💥  java:8     0.0.7      fdd0cddaa0ce443290cddaa0ce443288 cold  526  230  230ms    qr-java       
10/11 14:08:31  success  java:8     0.0.6      ebdb440fe04a47139b440fe04ab71395 cold  520  352  929ms    qr-java       
10/11 14:05:58  success  nodejs:10  0.0.5      112e26379bd546deae26379bd5f6ded0 warm  125  0    11ms     qr            
10/11 14:04:45  success  nodejs:10  0.0.5      b3c0a39480fa407a80a39480fae07a7f warm  142  0    8ms      qr            
10/11 14:04:26  success  nodejs:10  0.0.5      bfa6d1e3e40a404aa6d1e3e40af04a51 warm  151  0    32ms     qr  
```

```
> nim trigger list
Datetime        Status            Version  Trigger                                           
10/21 17:31:06  1 active          0.0.1    fire-each-minute     
```

```
> nim rule list
Datetime        Status            Version  Rules                                             
10/20 21:10:30  active            0.0.1    fire-each-minute-index  
```

```
> nim package list
Datetime        Access   Kind     Version  Packages                                          
10/21 18:32:32  private  binding  0.0.1    school-binding                                    
10/21 11:09:29  private  package  0.0.3    school    
```

**Comment by rabbah** [comment]
I've started to rebase to sync up with upstream. A wholesale rebase didn't work - too many tests failed. I am now cherry picking commits one at a time, and ensuring tests pass at each step. I am working on a new branch for now `dev-cherry-pick`.

There are at least 3 issues which will require additional work post rebase:
1. `activation log` enhancement we added at odds with aio
2. `route` override of `whisk.system` package name
3. `activation poll` did not work, requires some aio namespace setting which we don't use
4. test `trigger feed` support

Update: I've completed 25 of 42 cherry picks so far with all tests passing.

**Comment by rabbah** [comment]
Here are the remaining commits which I have not replayed (i.e., a check mark means the commit was cherry-picked and no check mark means it was not).

### header or cors issues
cc @joshuaauerbachwatson

The first one will require some tests to be adapted. The second causes a failure in the aio runtime library when the namespace is missing from the options.
  - [x] commit 2fcaef5b37373df77daed9e4753d5d9e4e996626
    Toggle extra logs on invoke. In support of nimbella-corp/cloud-workbench#2
    **NOTE:** This requires a change in the wb.
  - [x] commit 0b4654ad492d14dd99895e26609c1176b66b61f1
    Suppress namespace extra header when no namespace
    **NOTE:** I did rebase this to the new branch, but it will cause `activation` related commands to fail due to a missing namespace.

### config patch
This causes tests to fail @joshuaauerbachwatson 
  - [ ] commit 7e1a1fd3254861297acf5edc23c10c20aa537a51
    One possible way to fix config for browsers. (#4)
    Probably not the best way.  For nimbella-corp/cloud-workbench#1
  - [x] commit e685b17f66bace9642f0dcd9a3da437da6e9fcbb
    Allow properties file read to fail, eg, in browser
    **NOTE:** This commit causes coverage to drop but is otherwise ok.

---

### activation logs related
I will have to replay those more slowly.
  - [x] commit e73188cb61a6bf29df883e515a27c64318b06d7c
    Allow get logs --last to skip for added convenience. Add some parity for --logs with logs command. (#15)
  - [x] commit f1be9edcdefb4d827e9a50af2140a8bf5bf73b59
    If no activation id is given, assume --last.
  - [x] commit 2e45c318bea717f3c06c19ee3c7d19f1bce8f577
    Fetch multiple activation logs at once with activation filtering (#14)

Skipping these for now.
  - [ ] commit a6b3cf31ef9bba68967665460ff8468e51d34465
    Pretty print log lines as JSON where possible. (#14)
  - [ ] commit bd667ff2d8b4415dfd5ce2eec3d167fa3ba59d33
    Allow log lines to be slices and add flag to supress header. (#13)

----

### api gateway override
  - [x] commit 743ef106ce1542db2a2ce775a82da4d856db3ae9
    Override API gateway path.

### trigger feed
This should be subsumed by the rebase.
  - [ ] commit 9a45f491f0e0378174b87441b4bc9bd28576125c
    Support --feed on 'nim trigger create' (#2)

**Comment by joshuaauerbachwatson** [comment]
While I expect to do most of the work on this going forward, I'm assigning @rabbah in addition to myself to keep both of us in the loop.  I will probably need his help again.

A lot of the work will actually be in `nimbella-cli` and `cloud-workbench` but tracking it here because Rodric made a good start on laying out the issues and we must ultimately do the merge here.

The immediate issue is getting branches going in the other repos that incorporate `dev-cherry-pick` branch from this repo.   I've changed some mechanisms in `nimbella-cli` and have been able to do that.

The resulting branch doesn't build yet.  There are changes in how `aio-cli-plugin-runtime` exports its internal modules and some utilities have moved to another plugin.   Working on that first.

**Comment by joshuaauerbachwatson** [comment]
Branches `test-new-aio` in public and private `nimbella-cli` are now building and the deployer tests are passing.   This does not tell us all we need to know yet because (1) I haven't tried building the workbench yet and (2) the deployer tests mostly exercise the deployer (plus `nim action [ list | get | invoke ]`) but don't exercise other aio functionality very much.

I will try branching the workbench next since several of the questions about omitting or changing commits are about the effect on the workbench.

We can also now put the `nim` branches through Jenkins, which may reveal a few more things (more usage of `aio`-based commands in some of the functional tests).

**Comment by joshuaauerbachwatson** [comment]
The workbench build has stumbled across some issues.   Working on them but meanwhile testing In Jenkins starting with build job 3627

**Comment by joshuaauerbachwatson** [comment]
I can get the workbench to build but I am presently stumped on how to get it to run.  There is a module-level filesystem behavior somewhere in the new `aio-lib-runtime` that is breaks in a browser.   It might have to do with their logging.  I haven't pinned it down yet.  It's been frustrating.

**Comment by joshuaauerbachwatson** [comment]
> testing In Jenkins starting with build job 3627

Completed in teardown job 1029 with no intervening failures.   That's the good news.  The workbench is the bad news.

**Comment by joshuaauerbachwatson** [comment]
Failures of workbench initialization have been tracked down to the `fs-extra` package, which is used by `aio-lib-runtime` in such a way that its module initializations will always execute.   Those module initializations include some that require a real file system and will fail in a browser.   It has been tedious to track these down but I believe it is still the easier course to find them and patch them out (as I have done in other cases) rather than solving the problem through the webpack config.  I think sooner or later we will have to bite the bullet and either clone-and-own the kui webpack builder or start patching its config.  I have avoided this since the kui webpack build is pretty complex.

I still don't know for sure whether `fs-extra` is the _only_ source of these breakages in the aio runtime 3.0 but I'm hopeful.

**Comment by joshuaauerbachwatson** [comment]
It looks like I found all the breakages in `fs-extra` (three needed correction).   

There are also breakages due to `aio-lib-runtime` usage of `globby` and (gasp) `webpack`.   Wouldn't you know, webpack has some problems webpacking itself (reminds me of "Being John Malkovich").   In this case, I didn't chase the problem into the dependencies.   In the case of `fs-extra` (whose use in aio is now pervasive) it was the less intrusive course.  But, for `globby` and `webpack`, which are only used in one place each (and not a place we care about), it was easier to relocate the dependency so the `require` no longer executes at module scope.

With those five places now patched, the cloud-workbench again initializes.  The workbench tests pass in chrome.  They fail in firefox (timeout while starting up).

Despite the test failures in firefox, the workbench seems to run fine in both firefox and safari (we have no Selenium tests for safari).    Clearly, the problem that motivated commit 7e1a1fd is gone and we can abandon that commit.   That it is no longer needed is plausible since the config plugin has been substantially revised, including some changes to regexes (which were the problem before).

**Comment by joshuaauerbachwatson** [comment]
The test failures on firefox seem timing-related and I'm not worrying about them at this point.  I replayed the initial sequence of the test manually in firefox and it works.

Concerning commits 2fcaef5 and 0b4654a, I'll work on those next.   I think commit e685b17 is less immediately critical because it affects only test coverage and not customer-visible behavior.

**Comment by joshuaauerbachwatson** [comment]
There was a glitch when it turned out that the latest commits on `dev-cherry-pick` in Rodric's fork were. not pushed to the `nimbella` fork.   I redid the tagging and rebuilt the branches in other downstream repos and retested.  No regressions, except that now we have commit 43e5b16, which creates a toggle for the old commit 2fcaef5.   This required a balancing change to `nim` (which is only made when running in a browser).  The change simply inverts the toggle for that case as Rodric suggested.

**Comment by joshuaauerbachwatson** [comment]
The replacement for the old namespace header suppression commit (0b4654a) is 5fd5484.   I've added a toggle for this case and adjusted it a couple of times but I think it's right now.  It is set to false normally (so the header is not suppressed, that is, it is sent).   This is apparently needed by Adobe.   It is not needed by us but the `nim` command leaves the toggle alone except when running in the workbench, so our CLI paths send it as well.

In `nim` when running in a the workbench (only), the toggle it is inverted so that the header is suppressed.   This is need to avoid CORS issues in preview versions of the workbench which we need to have working so we can test features.   It is not actually critical for the production workbench but not being able to test features is enough of an issue that I regard this change as necessary.

The mechanism for removing the header is to remove the namespace option from the OW options.   For activation logs (only, among the aio commands that we adopt) there is an options check in `aio-lib-runtime` that throws due to this omission.   So, in the workbench build (only) we remove that check using patch-package.

The `activation logs` command works correctly in the CLI and I believe it is actually doing the right thing in the wb too but the output does not show up.   This is a separate regression due to `aio` 3.0 that also needs resolution before we can merge.

The output change for activation logs in 3.0 is something to do with special routines in `aio-lib-runtime` that do log printing.   Most of `aio-cli-plugin-runtime` still does output the old way and that continues to work.   The routines use a logging method that is passed to them.   The interception I am doing in the workbench _should_ mean that this logging method is connected to the capture logger and so it should work but empirically it does not.   

Two things to investigate.
1.  The logger gets passed through several places and perhaps somewhere the binding of the method to the CaptureLogger object could get lost.   
2.  If anywhere in this code path the Promise chain gets detached, the return of the call stack to the kui dispatcher could race ahead of the output production.   One wouldn't notice this sort of race in the CLI.

Once that is fixed I think it's just cleaning up the aio tests and we can merge.   In fact, once we have "all clear" from nim and wb, we could merge and clean up the aio tests afterwards prior to the next round of sending mods upstream.

**Comment by joshuaauerbachwatson** [comment]
I found and fixed what looked like a possible culprit (promise race) but it didn't change the behavior.  A very annoying meta-issue is that debugging gets turned off abruptly during workbench initialization by something newly introduced in aio 3.0 (didn't happen before).   I think this is due the aio core logging plugin, which does an unconditional `debug.enable` using the value of DEBUG from the process environment.   Since a browser doesn't use the process environment DEBUG variable to control debugging it is always undefined.   Checking out that hypothesis via yet another patch (sigh).

**Comment by joshuaauerbachwatson** [comment]
I fixed the debugging which was important in itself.   I then added a lot of debugging output.  I learned that

- the capture logger is arriving intact where it is needed and so the hypothesis that it was somehow becoming unbound is rejected
- there is no race condition in promise handling as far as I can see

It's simply that the code behaves differently in the workbench versus the CLI for (I think) the same test case.  In the CLI, when the individual activations are fetched and scanned for logs, logs are found.  In the workbench, that scanning operation turns up no logs.  If I am somehow not using the identical test case in both places then I have been on a wild goose chase and there is no problem.   If that's not it, then I have to dig deeper to see why there is this divergence in behavior at the model level.   I no longer think there is output that is being swallowed.  There is simply no output being produced.

**Comment by joshuaauerbachwatson** [comment]
> If I am somehow not using the identical test case in both places then I have been on a wild goose chase

Yup.   It's true.   I'm a doofus.   I was not running the identical test case in both places.  I think I was more methodical initially, and I think it's probable that there was a promise race which I fixed with PR #20.   But, after that, while continuing to test, I started being careless about choice of tests and thought there was still a problem when there wasn't.

One source of confusion is the help text for the `--limit` flag.  It says: " Fetch the last LIMIT activation logs" which is misleading.   What it actually does is " Fetch the logs of the last LIMIT activations".   That was leading me to think that, in some cases, logs ought to be found when in fact they wouldn't be (because the last LIMIT activations had no logs, though older ones did).

I believe we can proceed to a merge in the following order

* [x] Merge PR #20 into `dev-cherry-pick`
    - I have been asking for reviews from @rabbah for these changes because I wanted to observe his principles about how to maintain that branch.  But, we already have one failing aio test (expectedly) and a coverage deficiency on the aio testing side, and I doubt he will ask me to fix that (he didn't the last two times).   So, think it is better to just go ahead with the plan.   Fixing the tests comes later.   I will do that unless Rodric actively objects soonish.
* [x] Merge the consuming branches in `nimbella/nimbella-cli`, `nimbella-corp/nimbella-cli` and `cloud-workbench`.
    - this could be done without waiting for the previous since there is already a tag I can use.  Depending on the remaining steps, we might need to redo the tagging one or two more times, which is no big deal.
* [x] Do the hard reset (or whatever) to effectively rename `dev-cherry-pick` to `dev`
    - I want Rodric to either do this or give me a syntactically precise cookbook ... it is outside my present comfort zone
    - if this orphans the old tag I will redo the tagging.  We want all tags to be on active branches to avoid confusion the next time around
* [x] Fix the `aio` tests
* [ ] Revisit the few commits that Rodric ignored when building `dev-cherry-pick`

**Comment by rabbah** [comment]
I merged PR 20 and squashed 3 commits related to suppressing of the namespace in certain contexts (into 1). These changes are on the dev-cherry-pick branch now.

**Comment by joshuaauerbachwatson** [comment]
I went ahead and merged the consuming branches.  Because of the way we use tags, this means that the change in `aio` is now _live_ for us, even though the tag is on the `dev-cherry-pick` branch.   

I decided to go ahead and merge ahead of running yet another Jenkins test.   Changes since the last successful Jenkins test have been minor (and mostly in the workbench, which won't be tested by that).   I'd rather go live (internally, since none of this is in stable versions yet) and find any problems through usage ahead of having some need to put out a stable version.

**Comment by joshuaauerbachwatson** [comment]
Rodric verified the build post-merge ... he got aio 3.0.   He also renamed the branches.   So, we're almost done.

**Comment by joshuaauerbachwatson** [comment]
The remaining problems under this issue are to fix the tests and restore some recent commits of Rodric's which got deferred in his initial push.

One test broke because it was designed to cover the old (unconditional) way that we handled omission of the namespace header.   It was easily fixed by setting the newly-added toggle as part of the test.   This led to "all tests passing" but not quite 100% coverage.   That much is committed.

The jest coverage report does not call out any lines but the problem is definitely in RuntimeBaseCommand.js and Rodric believes, plausibly, that it is due to the try/catch block we added to keep the property file loader from crashing the workbench.   The empty catch block is not covered by any test because the `propertiesFile()` call is designed not to fail and cannot be made to fail in a CLI without some mockery.   We could add that mockery but I think a better solution is to change the way we accomplish the goal here so that we don't have the try/catch block at all.  I can probably insert some "browser friendliness" directly into `properties.js` so that it is harmless in a browser and I think that's maybe a better model for the Adobe team if we decide to send this change upstream (the try/catch is pretty indirect and hard to explain whereas a direct "don't use the file system if you're in a browser" seems clearer and more maintainable going forward.

**Comment by joshuaauerbachwatson** [comment]
Rodric fixed the tests without changing away from the current try/catch logic.  We now have 100% coverage and I checked off that task.

The remaining task concerns two commits that never got cherry-picked.  Both are enhancements to `activation logs`.   The bulk of the logic in `activation logs` (including the output display) is now in `aio-lib-runtime` which we don't currently fork.    I am leaving it to another day to decide whether to fork that repo and add back our enhancements to `activation logs`.   I am retitling this issue to cover the residual task.

I am not un-assigning myself from the issue because I need to keep it on my board to remember that the change to `aio` 3.0 is a significant change affecting the next stable release.   When the next stable release happens, I will take my name off this issue.

**Comment by joshuaauerbachwatson** [comment]
It appears that the external kafka test breaks now because it depends on a using the `--last --count <n>`  feature of `nim activation result`.   This was regressed.   I don't understand why I didn't catch that when I ran the test (build 3627 -> teardown 1029).   Investigating.

**Comment by joshuaauerbachwatson** [comment]
Yes.  External kafka job 655, which was in that sequence, passed.  I checked that the correct branch was being tested in that sequence.  It was.  I don't know why it didn't fail then since it is failing now.   In the log of job 655 you see

```
nim activation result --last -c 10
```

and it works.   Starting with job 672 we see

```
 ›   Error: Unexpected argument: 10
```

in response.   I get the same response running the command locally.   The likely cause is that the count feature is part of one of the commits we deferred from the cherry pick, thinking that only Rodric was using the functionality.  😄 .   The fact that this wasn't caught earlier is completely unexplained but I'm not wasting more time trying to figure that out.

**Comment by rabbah** [comment]
This was one of the changes where I allowed AIO to take precedence over our change. The flag is now —limit. The —last is implied and may be omitted.

**Comment by rabbah** [comment]
Hmm just noticed there’s a difference when —last used: a banner is printed between each result. The behavior should be the same imo.

**Comment by joshuaauerbachwatson** [comment]
Thanks.   I was just looking at the omitted commits and realized I was on the wrong track.  So, we should be able to fix this by just changing the test and no change to aio should be needed.

**Comment by joshuaauerbachwatson** [comment]
Oh.  I missed your intervening comment.   So, probably the banner should be eliminated, right?

**Comment by joshuaauerbachwatson** [comment]
To be precise: we can fix the external kafka job just by changing the syntax in the job without changing aio as long as `--last` is not used explicitly (just tried this out).   For some reason (haven't read the code) explicit use of `--last` triggers this intervening banners business, which is weird and probably should be fixed but is orthogonal.

**Comment by joshuaauerbachwatson** [comment]
For now, I am assuming we will fix the test and not change aio at this time.   Leaving this issue open.  If we think more changes are needed we should probably retitle the issue again.

**Comment by rabbah** [comment]
I think one of the commits I dropped added a `--quiet` flag which omits the banner which I think is useful when looking at multiple results and trying to correlate them to an event.

**Comment by rabbah** [comment]
@joshuaauerbachwatson I propose closing this issue as "completed" good enough for now.

**Comment by joshuaauerbachwatson** [comment]
Agreed.
